oval_graph/oval_node.py

```diff
@@ -56,20 +57,23 @@ def __init__(self, **kwargs):
         self.children = []
         if input_children:
             for child in input_children:
-                self._add_child(child)
+                self.add_child(child)
 
-    def _validate_negation(self, input_negation):
+    @staticmethod
+    def _validate_negation(input_negation):
         if not isinstance(input_negation, bool):
             raise TypeError("Wrong value of negation argument!")
         return input_negation
 
-    def _validate_type(self, input_node_type):
+    @staticmethod
+    def _validate_type(input_node_type):
         node_type = input_node_type.lower()
         if node_type not in ("value", "operator"):
             raise TypeError("Wrong value of node_type argument!")
         return node_type
 
-    def _validate_value(self, input_value):
+    @staticmethod
+    def _validate_value(input_node_type, input_value):
         value = input_value.lower()
 
         allowed_values = [
```

Comment:
Just a note - those two data structures could be global sets (instead of local lists). The "in" operator works faster with sets.